app/src/org/commcare/android/javarosa/IntentCallout.java

```diff
@@ -303,10 +309,53 @@ private void storePointerToFileResponse(TreeReference ref, String responseValue,
         if (!src.exists()) {
             //TODO: How hard should we be failing here?
             Log.w(TAG, "CommCare received a link to a file at " + src.toString() + " to be included in the form, but it was not present on the phone!");
+
+            Uri uri = Uri.parse(responseValue);
+
+            try {
+                ParcelFileDescriptor inFile = androidContext.getContentResolver().openFileDescriptor(uri, "r");
+
+                File newFile = new File(destinationFile, uri.getLastPathSegment());
+
+                long fileLength = -1;
+
+                //Looks like our source file exists, so let's go grab it
+                try {
```

Comment:
would be good to abstract the logic here to handle the uri and copy it using FileChannel into a `FileUtil` function.

Reply:
Definitely. This is actually directly pulled from the fileutil (sloppily) and needs a refactor back into it. Will update.